src/interlocking_component/route_controller.py

```diff
@@ -106,16 +109,53 @@ class RouteController(IRouteController):
     It makes sure, that the Interlocking sets fahrstrassen along those routes.
     """
 
-    def is_new_fahrstrasse_needed(self, train: "Train", track_segment: "Track"):
+    interlocking: interlockinginterface = None
+    router: Router = None
+
+    def is_new_fahrstrasse_is_needed(self, train: Train, track: Track) -> bool:
```

Comment:
You have to `is` in the name of this function. Do you need it? 🙃

Reply:
No, I think you are right, that is not needed. This is the method, that checks and will handle if the fahrstrasse could not be set, so there is no reason to return if it was successful.